src/OSPSuite.Core/Domain/Services/ObservedDataTask.cs

```diff
@@ -18,9 +18,10 @@ public interface IObservedDataTask
 
       /// <summary>
       ///    Deletes the <paramref name="observedData" /> from the project.
+      ///    If <paramref name="promptBeforeDeleting"/> then prompts the user before deleting
       ///    Returns <c>true</c> if the deletion was confirm by the user otherwise <c>false</c>
       /// </summary>
-      bool Delete(DataRepository observedData);
+      bool Delete(DataRepository observedData, bool promptBeforeDeleting = true);
```

Comment:
I don't think this is right as the prompt will be asked  FOR each single data repo. I think the method should delete using the other method taking a IEnumerable. That's why I did not add the prompt on this method in the first place

Reply:
@msevestre I think it makes more sense to pass the IEnumerable but still this will not answer the original question. We would like to control from the caller whether we want to prompt the user or not "do you really want to delete..?" So in some cases it makes sense to ask but if you look into the issue discussion, Pavel and I agreed on Juri's possible course of action where we do not to ask in case of a reload since the user was already informed about what was going to be deleted. So from the user point of view it is like:
1. I will remove this, update this and add this -> ok/cancel
2. Do you really want to remove this? -> yes/no

I normally do not like when I am prompt that much about something and we have the history for undoing commands so the final proposal was not to prompt in this case. So, all in all, either we have a boolean promptBeforeDeleting or we have two methods one for prompting and another one without it. Sure, internally they both will use the same private method in case we have two public methods for this and the private will have then the flag.

All in all, I agree it is better to pass the whole IEnumerable and then the delete should prompt only once in case of a list but it should also have the option not to prompt at all.

Reply:
Wel the option with the IEnumeralbe also as the flag. So I do not see a contradiction here?

Reply:
https://github.com/Open-Systems-Pharmacology/OSPSuite.Core/blob/7ed7872ceeef5b204c5152374cd06634d38ee4be/src/OSPSuite.Core/Domain/Services/ObservedDataTask.cs#L30

Reply:
and it's called `silent`. We should at least align the name of the flags...

Reply:
SO my suggestion is till
1-Undo the changes
2-In the method, instead of calling Delete (obsData, prompt=false) for each obsData to delete, do Delete(ListOfObsData, silent=true)

Reply:
There is no contradiction. I was stressing that we need both things ;)